src/shared/multiline-strings.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Remove leading and trailing empty lines from a string.
+ */
+const trimEmptyLines = (str: string): string =>
+  str.replace(/^\s*\n|\n\s*$/g, '');
+
+/**
+ * Remove leading indentation from each line of a string.
```

Comment:
```suggestion
 * Remove common indentation from each line of a string.
```

I think this makes it clearer that only the minimal shared indentation is removed.